### PR TITLE
Use /dev/tty by default

### PR DESCRIPTION
**Pull request by eugene-eeo**
This is because vim (and other terminal editors) typically output to stdout. This is not very nice behaviour especially if the program using the editor is piped to some other program. Hence, using tty is saner. Example:

```python
# test.py
import sys
import editor

cont = editor.edit(contents='ABC!',
                   use_tty='use_tty' in sys.argv)
sys.stdout.write(cont)
```

```shell
$ python test.py > t.xt
Vim: Warning: Output is not to a terminal
```

```shell
$ python test.py use_tty > t.txt
# happily edit in vim
$ cat t.txt
ABC!
```

### CODE REVIEW

**Comment by fmoo** [comment]
Some initial feedback:

- can you test with gedit and gvim as your editor? I've been taking special care to make sure that continues to work. 
- I'd like to avoid picking up the dependency on /dev/ as it's not very portable.  Is there a more platform independent way of solving this?
- I'm not yet convinced this should be the default.  What do you think about comparing isatty on sys.stdin and sys.stdout and being a little more discriminatory as to when we enable the feature?

**Comment by fmoo** [comment]
Also, style wise, I prefer assignment and a light branch to ternary operators and most extra logic to be split into helper functions. 

Also prefer explicit kwargs to **extra.  E.g., explicitly pass stdout=... and initialize to sys.stdout I guess?

**Comment by eugene-eeo** [comment]
Tested on MacVim ('mvim') and Sublime Text as they are the only GUI editors I have on OSX, it doesn't work for both cases. The string gets returned immediately.

On `/dev/tty` - unfortunately such a thing doesn't exist on windows. However one can always run cygwin, and given that it is used in many other tools (selecta comes to mind), I *expect* it to be pretty much universal for modern distros. From Unix Stackexchange:

 > `/dev/tty` works as long as the process has a controlling terminal

We can check if `std{out,in,err}` are ttys before we enable the feature as well, if you want.

File: test.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import sys
+import editor
+
+cont = editor.edit(contents='ABC!',
+                   use_tty='use_tty' in sys.argv)
+sys.stdout.write(cont)
```
Comment (fmoo) [inline test.py]: Please add the blank newline at the end of file

**Comment by fmoo** [comment]
> unfortunately such a thing doesn't exist on windows. 

I think it does, via the special file "CON:".  I'll have to test when I'm on windows later today.

The default should be `use_tty=None`.  Then:

```
if use_tty is None:
    use_tty = sys.stdin.isatty() and not sys.stdout.isatty()
```

Add a helper function to get the tty filename:
```
def get_tty_filename():
    if sys.platform == 'win32':
        return 'CON:'
    return '/dev/tty'
```




Which is exactly why the current

**Comment by fmoo** [comment]
Pushed to pypi as 1.0

**Pull request merged**